scripts/test_solidity_lsp.py

```diff
@@ -0,0 +1,624 @@
+                'text': self.get_test_file_contents(FILE_NAME)
+            }
+        })
+        published_diagnostics = self.wait_for_diagnostics(solc, 1)
+        solc.send_message('textDocument/didChange', {
+            'textDocument': {
+                'uri': FILE_URI
+            },
+            'contentChanges': [
+                {
+                    'range': {
+                        'start': { 'line': 3, 'character': 7 },
+                        'end': { 'line': 3, 'character': 7 }
+                    },
+                    'text': " C"
+                }
+            ]
+        })
+        published_diagnostics = self.wait_for_diagnostics(solc, 1)
+        self.expect_equal(len(published_diagnostics), 1)
+        report1 = published_diagnostics[0]
+        self.expect_equal(report1['uri'], FILE_URI, "Correct file URI")
+        self.expect_equal(len(report1['diagnostics']), 1, "one diagnostic")
+        # TODO: not done yet. we're getting the wrong diagnostic back (as if we didn't edit)
+
+    def test_textDocument_didChange_empty_file(self, solc: JsonRpcProcess) -> None:
+        """
+        Starts with an empty file and changes it to look like
+        the didOpen_with_import test case. Then we can use
+        the same verification calls to ensure it worked as expected.
+        """
+        # This FILE_NAME must be alphabetically before lib.sol to not over-complify
+        # the test logic in verify_didOpen_with_import_diagnostics.
+        FILE_NAME = 'a_new_file'
+        FILE_URI = self.get_test_file_uri(FILE_NAME)
+        self.setup_lsp(solc)
+        solc.send_message('textDocument/didOpen', {
+            'textDocument': {
+                'uri': FILE_URI,
+                'languageId': 'Solidity',
+                'version': 1,
+                'text': ''
+            }
+        })
+        reports = self.wait_for_diagnostics(solc, 1)
+        self.expect_equal(len(reports), 1)
+        report = reports[0]
+        published_diagnostics = report['diagnostics']
+        self.expect_equal(len(published_diagnostics), 2)
+        self.expect_diagnostic(published_diagnostics[0], code=1878, lineNo=0, startColumn=0, endColumn=0)
+        self.expect_diagnostic(published_diagnostics[1], code=3420, lineNo=0, startColumn=0, endColumn=0)
+        solc.send_message('textDocument/didChange', {
+            'textDocument': {
+                'uri': self.get_test_file_uri('a_new_file')
+            },
+            'contentChanges': [
+                {
+                    'range': {
+                        'start': { 'line': 0, 'character': 0 },
+                        'end': { 'line': 0, 'character': 0 }
+                    },
+                    'text': self.get_test_file_contents('didOpen_with_import')
+                }
+            ]
+        })
+        published_diagnostics = self.wait_for_diagnostics(solc, 2)
+        self.verify_didOpen_with_import_diagnostics(published_diagnostics, 'a_new_file')
+
```

Comment:
More test ideas:

1.
 - didOpen on two files that import each other but do not exist in the file system
 - didClose the imported one - it should lead to an error in the other (because it does not exist)

2.
 - didOpen on an existing file. Change it to add an error. didClose should revert to the version on disk and thus remove the error.